File: app/src/main/java/knf/kuma/tv/cards/AnimeCardView.java
package knf.kuma.tv.cards;

import android.content.Context;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.commons.PicassoSingle;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.tv.BindableCardView;

public class AnimeCardView extends BindableCardView<AnimeObject> {

    @BindView(R.id.img)
    ImageView imageView;
    @BindView(R.id.title)
    TextView title;

    public AnimeCardView(Context context) {
        super(context);
        ButterKnife.bind(this);
    }

    @Override
    public void bind(AnimeObject data) {
        PicassoSingle.get(context).load(data.img).into(imageView);
        title.setText(data.name);
    }

    @Override
    public ImageView getImageView() {
        return imageView;
    }

    @Override
    public int getLayoutResource() {
        return R.layout.item_tv_card;
    }
}

File: app/src/main/java/knf/kuma/tv/cards/AnimeCardView.kt
package knf.kuma.tv.cards

import android.content.Context
import android.widget.ImageView
import android.widget.TextView
import knf.kuma.R
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.load
import knf.kuma.tv.BindableCardView
import knf.kuma.tv.search.BasicAnimeObject
import org.jetbrains.anko.find

class AnimeCardView(context: Context) : BindableCardView<BasicAnimeObject>(context) {

    override val imageView: ImageView
        get() = find(R.id.img)
    override val layoutResource: Int
        get() = R.layout.item_tv_card

    override fun bind(data: BasicAnimeObject) {
        imageView.load(PatternUtil.getCover(data.aid))
        find<TextView>(R.id.title).text = data.name
    }
}
